contrib/pkg/broker/user_provided/controller/controller.go

```diff
@@ -61,8 +72,13 @@ func (c *userProvidedController) Catalog() (*brokerapi.Catalog, error) {
 	}, nil
 }
 
-func (c *userProvidedController) CreateServiceInstance(id string, req *brokerapi.CreateServiceInstanceRequest) (*brokerapi.CreateServiceInstanceResponse, error) {
+func (c *userProvidedController) CreateServiceInstance(
+	id string,
+	req *brokerapi.CreateServiceInstanceRequest,
+) (*brokerapi.CreateServiceInstanceResponse, error) {
 	credString, ok := req.Parameters["credentials"]
+	c.rwMutex.Lock()
+	defer c.rwMutex.Unlock()
```

Comment:
@duglin yes, we could move the mutex closer to the mutation. Doing so adds a bit more code -- and thus harder-to-read code -- to the broker, which is not performance critical. Until we discover that (a) there's a bottleneck on this or other similar function calls and (b) the broker throughput is too low, this is a premature optimization IMO